### PR TITLE
feat: Query sharing s3

### PR DESCRIPTION
**Pull request by gaurpulkit**
- Modifying the existing Query Result Sharing PR by using a pre-signed s3 url to send the data directly to the user's s3 server instead of storing the data in Altimate's database.

### CODE REVIEW

**Comment by mdesmet** [comment]
I think the point is to have both ways. We should offer the choice to the user to either use s3 bucket or use our saas.

@anandgupta42: Can you confirm?

File: src/webview_provider/queryResultPanel.ts [hunk 1]
```diff
@@ -219,25 +219,53 @@ export class QueryResultPanel implements WebviewViewProvider {
             window.withProgress(
               {
                 location: ProgressLocation.Notification,
-                title: "Uploading data to Altimate backend",
+                title: "Generating Shareable URL",
```
Comment (saravmajestic) [inline src/webview_provider/queryResultPanel.ts:222]: add telemetry

File: src/webview_provider/queryResultPanel.ts [hunk 2]
```diff
@@ -219,25 +219,53 @@ export class QueryResultPanel implements WebviewViewProvider {
...
                     window.showInformationMessage(
-                      `URL is copied to clipboard: [${result.share_url}](${result.share_url})`,
+                      "Generated signed url. Uploading data...",
                     );
+                    const uploadResponse =
+                      await this.altimate.uploadDataToSignedUrl(
+                        result.signed_url,
+                        message,
+                      );
+                    if (uploadResponse.status === 200) {
```
Comment (saravmajestic) [inline src/webview_provider/queryResultPanel.ts:237]: early returns

File: src/webview_provider/queryResultPanel.ts [hunk 3]
```diff
@@ -219,25 +219,53 @@ export class QueryResultPanel implements WebviewViewProvider {
...
+                      const verifyResponse =
+                        await this.altimate.verifyShareQueryUpload({
+                          sharing_table_id: result.sharing_table_id,
+                        });
+
+                      const verifyData = verifyResponse as {
+                        ok: boolean;
+                        share_url: string;
+                      };
+                      if (verifyData.ok) {
```
Comment (saravmajestic) [inline src/webview_provider/queryResultPanel.ts:248]: early returns

**Review by saravmajestic — CHANGES_REQUESTED** [review]
some cleanup needed

File: src/webview_provider/queryResultPanel.ts (hunk 2 above)
Comment (mdesmet) [reply src/webview_provider/queryResultPanel.ts:237]: Also http related stuff shouldn't leak into implementations.

**Comment by saravmajestic** [comment]
Not needed anymore

**Pull request closed without merging**